### PR TITLE
Micro opt base64 decode

### PR DESCRIPTION
**Pull request by juj**
Micro-optimize code size of base64 decoding via the observation that 'true/false' coerce to 1/0 in arithmetic, e.g. `3 - true == 2`.

### CODE REVIEW

File: tests/code_size/hello_webgl2_wasm.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-  "total": 16566,
-  "total_gz": 9746
+  "a.html.gz": 377, 
+  "a.wasm.gz": 6940, 
+  "total_gz": 9739, 
+  "a.wasm": 10912, 
+  "a.js.gz": 2422, 
+  "a.js": 5068, 
+  "a.html": 563, 
+  "total": 16543
```
Comment (kripken) [inline tests/code_size/hello_webgl2_wasm.json:9]: not a big deal, but the order of the keys here changed, which makes it less easy to compare. Are you using a different python, or on a different platform for this commit? or maybe just noise.

might be worth sorting these.

Comment (juj) [reply tests/code_size/hello_webgl2_wasm.json]: Ugh, migrating to the file-based expectations with the EMTEST_REBASELINE mechanism is making updating test expectations extremely painful, as my results do not agree with CI results, and the CI aborts on first failure, whereas it used to run all the checks in the old form. And the sorting.

I'll revert and do this by hand one by one as CI yells.

Comment (kripken) [reply tests/code_size/hello_webgl2_wasm.json]: We should investigate why your results differ from CI. Are you on `emsdk tot`, and have cleared the cache locally?

Comment (juj) [reply tests/code_size/hello_webgl2_wasm.json]: I am running against latest master built from source, and yeah, have cleared the cache. The last time I looked at the deltas against CI, they were subtle differences in bytecode output, and/or closure minification naming choices. I don't quite have cycles to debug this further though :/

Comment (sbc100) [reply tests/code_size/hello_webgl2_wasm.json]: > Ugh, migrating to the file-based expectations with the EMTEST_REBASELINE mechanism is making updating test expectations extremely painful, as my results do not agree with CI results, and the CI aborts on first failure, whereas it used to run all the checks in the old form. And the sorting.
> 
> I'll revert and do this by hand one by one as CI yells.

The sorting this should be easy to fix.

Regarding the "aborts on first failure" thing?  I've not looked at the test in question, but that sounds like the test in question might not be fine grained enough.  Maybe we split it up, for example, by using @paramaterize on this test?  Ideally each test should only have a single way to if possible.

**Pull request merged**